Mathlib/RingTheory/Algebraic.lean

```diff
@@ -616,3 +721,92 @@ theorem Polynomial.algebraMap_pi_self_eq_eval :
   rfl
 
 end Pi
+
+namespace MvPolynomial
+
+variable {σ : Type*} (R : Type*) [CommRing R]
+
+-- TODO: move to suitable place
+private theorem rename_polynomial_aeval_X
+    {σ τ R : Type*} [CommSemiring R] (f : σ → τ) (i : σ) (p : R[X]) :
+    rename f (Polynomial.aeval (X i) p) = Polynomial.aeval (X (f i) : MvPolynomial τ R) p := by
+  rw [← AlgHom.comp_apply]
+  congr 1; ext1; simp
+
+theorem transcendental_supported_polynomial_aeval_X {i : σ} {s : Set σ} (h : i ∉ s)
+    {f : R[X]} (hf : Transcendental R f) :
+    Transcendental (supported R s) (Polynomial.aeval (X i : MvPolynomial σ R) f) := by
```

Comment:
> I think this can be decomposed into two facts:
> 
> 1. if `i : R →+* A` makes A a flat R-algebra, then `f : R[X]` is transcendental over R implies that `f.map i : A[X]` is transcendental over A. (The converse holds when `i` is injective. More generally, R[X] can be replaced by any R-algebra and A[X] by its tensor product with A.) In the current situation, `A = R[s] = supported R s` is free over R, so we can avoid importing flatness.

Yeah, that was what I said in https://leanprover.zulipchat.com/#narrow/channel/144837-PR-reviews/topic/Transcendental.20and.20algebraically.20independent/near/480005599 . Note that the current proof in this PR does not require flatness or freeness, as it rearranges the aeval map as composition of several maps, and uses the fact that `[Mv]Polynomial.map` preserves injectivity.

> 2. If `f : A →ₐ[R] B` is injective, then `x : A` is transcendental over R iff `f x` is. This is equivalent to [isAlgebraic_algHom_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Algebraic.html#isAlgebraic_algHom_iff), and in the current situation there is an injective R[s]-AlgHom from R[s][X] to R[σ] that sends `f.map (algebraMap R R[s])` to `Polynomial.aeval (X i : MvPolynomial σ R) f`.

That is `AlgebraicIndependent.ringHom_of_comp_eq` etc. in PR #18648. I haven't added `AlgHom` version yet.